pact-lang-api.js

```diff
@@ -292,20 +296,70 @@ var mkExp = function(pgmName) {
  * @return {object} of arguments, type-checked and properly named.
  */
 var mkMeta = function (sender, chainId, gasPrice, gasLimit) {
-  if (typeof sender !== 'string' ) {
-    throw new TypeError('sender must be a string: ' + JSON.stringify(sender));
-  }
-  if (typeof chainId !== 'string') {
-    throw new TypeError('chainId must be a string: ' + JSON.stringify(chainId));
-  }
-  if (typeof gasPrice !== 'number' ) {
-    throw new TypeError('gasPrice must be a number: ' + JSON.stringify(gasPrice));
-  }
-  if (typeof gasLimit !== 'number' ) {
-    throw new TypeError('gasLimit must be a number: ' + JSON.stringify(gasLimit));
-  }
+  enforceType(sender, "string", "sender");
+  enforceType(chainId, "string", "chainId");
+  enforceType(gasPrice, "number", "gasPrice");
+  enforceType(gasPrice, "number", "gasLimit");
   return {"gasLimit":gasLimit, "chainId":chainId, "gasPrice":gasPrice, "sender":sender}
-}
+};
+
+/**
+ * Formats ExecCmd into api request object
+ */
+var mkReq = function (cmd) {
+  return {
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    method: 'POST',
+    body: JSON.stringify(cmd),
+  }
+};
+
+/**
+ * Sends Pact command to a running Pact server and retrieves tx result.
+ * @param {Object}
+ * @property pactCode {string} - pact code to execute
+ * @property keyPairs {array or object} - array or single ED25519 keypair
+ * @property nonce {string} - nonce value, default at current time
+ * @property envData {object} - JSON message data for command, default at empty obj
+ * @property meta {object} - meta information, see mkMeta
+ * @property apiHost {string} - host running Pact server
+ * @return {object} - tx result received from pact server.
+ */
+const sendCommand = async function ({pactCode, keyPairs, nonce=new Date().toISOString(), envData={}, meta=mkMeta("","",0,0), apiHost}) {
```

Comment:
Let's avoid repeating all the arguments of `simpleExecCommand` and instead just take the result from that as an `object` argument.

Reply:
Also let's break this into two functions, one for just the send `await fetch`, then one for listen, and let's add one for poll.

Reply:
- Separated send/listen, added poll. -  fetchSend, fetchLoccal, fetchListen, fetchPoll. 
- About arguments, this will be the first function that users call, which then calls simpleExecCommand with its arguments.